### PR TITLE
A CLI command to validate repo configs 

### PR DESCRIPTION
**Pull request by nikololiahim**
Attempt to solve #2638. Can be invoked like this (as a subcommand):
```
java -jar scala-steward-assembly-VERSION.jar validate-repo-config .scala-steward.conf
```
Sample input (with error):
`.scala-steward.conf`:
```ini
updatePullRequests = 123
```

Output:
```python
2022-07-12 11:17:08,589 ERROR Configuration file at /home/nikololiahim/scala/scala-steward/.scala-steward.conf contains errors:
  Decoding failed with:
    Got value '123' with wrong type, expecting string:
      updatePullRequests
```

### CODE REVIEW

**Comment by nikololiahim** [comment]
Now that I think of it, this doesn't solve the "actually valid config" problem described [here](https://github.com/typelevel/steward/pull/7#discussion_r903257737). It still just runs the file through a circe parser, and doesn't perform any additional checks.

**Comment by armanbilge** [comment]
In case you haven't seen it, a scala-cli version of this idea was proposed in https://github.com/typelevel/steward/pull/7#discussion_r903318794. Not sure if that's basically what you were doing here already.

**Comment by nikololiahim** [comment]
> In case you haven't seen it, a scala-cli version of this idea was proposed in [typelevel/steward#7 (comment)](https://github.com/typelevel/steward/pull/7#discussion_r903318794). Not sure if that's basically what you were doing here already.

Yes, I have seen this comment, and this is basically what I am doing here. I'm just wondering if this solution is good enough (or even necessary at all).

**Comment by armanbilge** [comment]
> I'm just wondering if this solution is good enough (or even necessary at all).

Not entirely sure what you mean, but it's definitely necessary :) basically it helps validate two things:
1. the config file is valid HOCON format (extraneous commas have messed us up multiple times)
2. furthermore, it's consistent with the Steward config "schema"

Of course, this won't guarantee that it "works", but it's a huge improvement over the status quo.

See also: https://github.com/scala-steward-org/scala-steward/pull/2446.

**Comment by nikololiahim** [comment]
So I tried to use the `--validate-repo-config` as it was prior to https://github.com/scala-steward-org/scala-steward/pull/2662/commits/7be5f91fa90f2879ee1a023ee1f1e8e127b030df and it turns out there are several required flags which are not relevant to the config validation part. So, I decided that it would be better to make a separate `validate-repo-config`. It can now be invoked like this:
```
java -jar scala-steward-assembly-VERSION.jar validate-repo-config .scala-steward.conf
```

**Comment by armanbilge** [comment]
I was editing a `.scala-steward.conf` a little earlier and missed this :) I hope a maintainer may be able to take a look soon, thanks!

**Comment by nikololiahim** [comment]
@armanbilge yeah, it's basically ready (at least functionality-wise), just needs a review

File: docs/help.md [hunk 1]
```diff
@@ -3,7 +3,9 @@
 All command line arguments for the `scala-steward` application.
 
 ```
-Usage: scala-steward --workspace <file> --repos-file <file> [--git-author-name <string>] --git-author-email <string> [--git-author-signing-key <string>] --git-ask-pass <file> [--sign-commits] [--vcs-type <vcs-type>] [--vcs-api-host <uri>] --vcs-login <string> [--do-not-fork] [--add-labels] [--ignore-opts-files] [--env-var <name=value>]... [--process-timeout <duration>] [--whitelist <string>]... [--read-only <string>]... [--enable-sandbox | --disable-sandbox] [--max-buffer-size <integer>] [--repo-config <uri>]... [--disable-default-repo-config] [--scalafix-migrations <uri>]... [--disable-default-scalafix-migrations] [--artifact-migrations <uri>]... [--disable-default-artifact-migrations] [--cache-ttl <duration>] [--bitbucket-server-use-default-reviewers] [--gitlab-merge-when-pipeline-succeeds] [--github-app-id <integer> --github-app-key-file <file>] [--url-checker-test-url <uri>] [--default-maven-repo <string>] [--refresh-backoff-period <duration>]
+Usage:
+    scala-steward validate-repo-config
```
Comment (exoego) [inline docs/help.md:7]:
```suggestion
    scala-steward validate-repo-config path/to/.scala-steward.conf
```

File: modules/core/src/main/scala/org/scalasteward/core/application/Context.scala [hunk 2]
```diff
@@ -77,12 +81,31 @@ final class Context[F[_]](implicit
...
+        val validateRepoConfigAlg: ValidateRepoConfigAlg[F],
+        val logger: Logger[F]
+    ) extends StewardContext[F] {
+      override def runF: F[ExitCode] = validateRepoConfigAlg.validateAndReport(file)
+    }
+  }
+
+  def step0[F[_]](
+      usage: Config.StewardUsage
+  )(implicit asyncF: Async[F]): Resource[F, StewardContext[F]] =
```
Comment (exoego) [inline modules/core/src/main/scala/org/scalasteward/core/application/Context.scala:103]: Could you revert and keep original `(implicit F: Async[F])`?
It is a convention to declare `implicit F: ~[F]` in the existing code base.

File: modules/core/src/main/scala/org/scalasteward/core/repoconfig/ValidateRepoConfigAlg.scala [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import org.scalasteward.core.io.FileAlg
+import org.scalasteward.core.repoconfig.RepoConfigAlg
+
+import ValidateRepoConfigAlg._
+import io.circe.{CursorOp, DecodingFailure, ParsingFailure}
+
+final class ValidateRepoConfigAlg[F[_]](implicit
+    fileAlg: FileAlg[F],
+    logger: Logger[F],
+    monadThrowF: MonadThrow[F]
```
Comment (exoego) [inline modules/core/src/main/scala/org/scalasteward/core/repoconfig/ValidateRepoConfigAlg.scala:33]:
```suggestion
    F: MonadThrow[F]
```
Naming convention in this repo.

File: modules/core/src/main/scala/org/scalasteward/core/application/Context.scala [hunk 4]
```diff
@@ -94,22 +117,46 @@ object Context {
...
+      }
+
     } yield context
 
+  def initRegular[F[_]](config: Config)(implicit
+      client: Client[F],
+      urlCheckerClient: UrlCheckerClient[F],
+      fileAlg: FileAlg[F],
+      logger: Logger[F],
+      asyncF: Async[F]
```
Comment (exoego) [inline modules/core/src/main/scala/org/scalasteward/core/application/Context.scala:144]:
```suggestion
      F: Async[F]
```
Naming convention in this repo.

File: modules/core/src/main/scala/org/scalasteward/core/application/Context.scala [hunk 5]
```diff
@@ -94,22 +117,46 @@ object Context {
...
+  ): Resource[F, Context[F]] = {
+    implicit val processAlg = ProcessAlg.create(config.processCfg)
+    implicit val workspaceAlg = WorkspaceAlg.create(config)
+    Resource.eval(step1(config))
+  }
+
+  def initValidateRepoConfig[F[_]](file: File)(implicit
+      fileAlg: FileAlg[F],
+      logger: Logger[F],
+      asyncF: MonadThrow[F]
```
Comment (exoego) [inline modules/core/src/main/scala/org/scalasteward/core/application/Context.scala:154]:
```suggestion
      F: MonadThrow[F]
```
Naming convention in this repo.

File: modules/core/src/main/scala/org/scalasteward/core/application/Context.scala (hunk 2 above)
Comment (nikololiahim) [reply modules/core/src/main/scala/org/scalasteward/core/application/Context.scala:103]: Yeah, sure, it's just that the pre-CI linter complained about it, I thought that was the issue somehow.

**Comment by exoego** [comment]
Thanks for your contribution.
I've added change requests regarding the naming convention.
Other parts look great to me.

**Comment by nikololiahim** [comment]
@exoego Thank you! Will fix the comments shortly.

File: docs/help.md (hunk 1 above)
Comment (nikololiahim) [reply docs/help.md:7]: I think this file is [auto-generated](https://github.com/scala-steward-org/scala-steward/blob/main/modules/docs/mdoc/help.md) from `decline`'s `Command` instance, which unfortnately does not generate placeholders for the arguments of subcommands. The help mesage can still be obtained via 
```
scala-steward validate-repo-config --help
```

**Comment by nikololiahim** [comment]
@exoego https://github.com/scala-steward-org/scala-steward/pull/2662/commits/5568b6f21d7be97a07eaac6c0e80f795372e034e addresses all the naming comments.

File: docs/help.md (hunk 1 above)
Comment (exoego) [reply docs/help.md:7]: I see

**Review by exoego — APPROVED** [review]

**Comment by exoego** [comment]
Sorry for late review

**Comment by exoego** [comment]
I am fine with a little coverage down and [minor Codacy issues](https://app.codacy.com/gh/scala-steward-org/scala-steward/pullRequest?prid=9766981).
https://app.codacy.com/gh/scala-steward-org/scala-steward/pullRequest?prid=9766981

**Pull request merged**